csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/InferenceTest.cs

```diff
@@ -2025,6 +2028,101 @@ public SkipNonPackageTests()
                 }
             }
         }
+
+        private static void AsyncCallback(IntPtr user_data, IReadOnlyCollection<NamedOnnxValue> outputs)
+        {
+            GCHandle evtHdl = GCHandle.FromIntPtr(user_data);
+            var evt = (ManualResetEvent)evtHdl.Target;
+            var valueOut = outputs.ElementAt(0);
+            var tensorOut = valueOut.AsTensor<Float16>();
+            var dims = tensorOut.Dimensions;
+            Assert.Equal(new Float16(16896), tensorOut.ElementAtOrDefault(2));
+            evt.Set();
+        }
```

Comment:
Naively I would have expected a C# user to not need any low level code dealing with IntPtr etc. Instead they would call an async Run that returns `Task<IReadOnlyCollection<NamedOnnxValue>>` and we would do the plumbing internally to facilitate that.

Can we use the new OrtValue instead of NamedOnnxValue? I believe @yuslepukhin has said that should be preferred over NamedOnnxValue.

Reply:
Switched to OrtValue, and no more IntPtr handling in customer code.